lib/colord/cd-icc.c

```diff
@@ -1677,6 +1680,47 @@ cd_icc_save_data (CdIcc *icc,
 			goto out;
 	}
 
+	/* finally, optionally override the created date */
+	if (priv->creation_time != -1) {
+		/*
+		 * LCMS2 maintainer rejects adding API to do this,
+		 * instead suggesting you munge the header directly:
+		 * https://github.com/mm2/Little-CMS/issues/71
+		 */
+		struct tm creation_time;
+		cmsICCHeader *header;
+		g_autoptr (GByteArray) mutable_data;
+
+		data = cd_icc_serialize_profile (icc, error);
+		if (!data) {
+			goto out;
+		}
+		mutable_data = g_bytes_unref_to_array (data);
+		data = NULL;
+
+		if (!gmtime_r(&priv->creation_time, &creation_time)) {
+			g_set_error (error,
+				     CD_ICC_ERROR,
+				     CD_ICC_ERROR_FAILED_TO_SAVE,
+				     "failed to translate creation time: %s (%i)",
+				     g_strerror(errno),
+				     errno);
+			goto out;
+		}
+		header = (cmsICCHeader*)mutable_data->data;
+		_cmsEncodeDateTimeNumber(&header->date, &creation_time);
+
+		/*
+		 * Now that we've changed the creation date, we need to recalculate MD5.
+		 * The simplest way to do that is to re-read the exported profile,
+		 * then calculate MD5 as normal.
+		 */
+		cmsCloseProfile (priv->lcms_profile);
+		priv->lcms_profile = cmsOpenProfileFromMemTHR (priv->context_lcms,
```

Comment:
Does this break any `CdTransform`'s created with the profile? i.e. do the tests pass?